Quick Start: New Visuals v2 with fix

momo-ozawa opened the pull request:
## Context
- In https://github.com/wordpress-mobile/WordPress-iOS/pull/18856 we reverted https://github.com/wordpress-mobile/WordPress-iOS/pull/18747, as we didn't have enough time to investigate / fix the root issue for some UI issues found during the CfT for WPiOS 20.0 
- This PR adds back the changes we made in https://github.com/wordpress-mobile/WordPress-iOS/pull/18747 and also fixes the issues mentioned on the CfT
  - Incorrect chevron icon aspect ratio (ref: p5T066-3gF-p2#comment-12523)
  - Ellipsis icon being displayed for the old tours (ref: p5T066-3gF-p2#comment-12533

## How to test


### With Menu as default

1. Run the app
2. Enable QS for existing site
3. ✅ Check that it appears correctly and it can be dismissed
4. Enable QS for new site
4. ✅ Check that it appears correctly and it can be dismissed 

### With Home as default

1. Run the app
2. Enable QS for existing site
3. ✅ Check that it appears correctly and it can be dismissed
4. Enable QS for new site
4. ✅ Check that it appears correctly and it can be dismissed
5. ✅ Check that the chevron icon on card appears correctly

Tab | QS for New Site | QS for Existing Site
-- | -- | --
Home | <img src="https://user-images.githubusercontent.com/6711616/174100630-8334e4e3-fa49-4a77-b11d-a73432a4ad7a.png" width=200> | <img src="https://user-images.githubusercontent.com/6711616/174100685-cdc2bea2-984a-4d37-b227-2657a00695ee.png" width=200>
Menu | <img src="https://user-images.githubusercontent.com/6711616/174100728-d6656a66-f1b0-470b-87b7-80459cf5cbc2.png" width=200> | <img src="https://user-images.githubusercontent.com/6711616/174100777-c5ad6ee2-5664-418b-8796-9879b1eadae6.png" width=200>


## Regression Notes
1. Potential unintended areas of impact
- Quick Start cards on dashboard / menu

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Tested manually

3. What automated tests I added (or what prevented me from doing so)
- N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Refactor: hide header for quick start card

Commit:
Refactor: extract stackview constarint configuration into separate methods

Commit:
Fix: UITableViewHeaderFooterView background color warning

Commit:
Fix: update source rect for QS card ellipsis button

Commit:
Fix: card frame view configuration for quick start type

Commit:
Fix: adjust margins for quick start cell on menu tab

hassaanelgarem commented on WordPress/Classes/ViewRelated/Blog/QuickStartCell.swift:
```diff
@@ -22,6 +22,16 @@ import UIKit
+        static func margins(for quickStartType: QuickStartType) -> UIEdgeInsets {
+            switch quickStartType {
+            case .undefined:
+                return .zero
```
Undefined here means this quick start started on an older version, which basically means the type is `newSite`. I think we should it treat it here this way.

momo-ozawa replied:
Good catch, thanks! Fixed -- f7ac1ba

hassaanelgarem commented on WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Quick Start/DashboardQuickStartCardCell.swift:
```diff
@@ -51,6 +43,38 @@ final class DashboardQuickStartCardCell: UICollectionViewCell, Reusable, BlogDas
     }
+
+    private func configureCardFrameView(for blog: Blog) {
+        switch blog.quickStartType {
```
Same comment here for handling `undefined` as if it's `newSite`

hassaanelgarem replied:
We should probably do the same for `String.title(for:)`. This is probably an existing bug 🙈

momo-ozawa replied:
Fixed -- f7ac1ba

hassaanelgarem reviewed (commented):
All looks good to me 🚀 
Just left a few comments, all related to the same thing.

Commit:
Fix: handle undefined quick start type

The `undefined` quick start type means that Quick Start was triggered on an older version. As such, `undefined` should be treated the same as `newSite`

momo-ozawa commented:
@osullivanchris Managed to find a solution to the UI bug 😄

osullivanchris commented:
> @osullivanchris Managed to find a solution to the UI bug 😄

Nice job! Downloaded the build and looks resolved from my side 👍 🙌
